Assets/SEE/Net/Network.cs

```diff
@@ -629,153 +573,201 @@ private void OnSceneUnloaded(Scene scene)
+        /// Sends the client back to the main menu because the connection could not
+        /// be established.
+        /// The <paramref name="owner"/> is not used.
+        /// </summary>
+        /// <param name="owner">ID of the owner (ignored)</param>
+        private void OnClientDisconnectCallback(ulong owner)
+        {
+            callbackToMenu?.Invoke(false,
+                                   $"The server {ServerAddress} has refused the connection due to the following reason: "
+                                     + NetworkManager.Singleton.DisconnectReason);
+            callbackToMenu = null;
+        }
 
         /// <summary>
-        /// Starts a dedicated server without client.
+        /// Starts a client.
         ///
         /// Note: This method starts a co-routine and then returns to the caller immediately.
         /// The <paramref name="callBack"/> tells the caller that the co-routine has come to
         /// an end.
         /// </summary>
         /// <param name="callBack">a callback to be called when done; its parameter will be true
         /// in case of success or otherwise false</param>
-        public void StartServer(CallBack callBack)
+        public void StartClient(CallBack callBack)
         {
-            StartCoroutine(ShutdownNetwork(InternalStartServer));
+            callbackToMenu = callBack;
+            StartCoroutine(ShutdownNetwork(InternalStartClient));
 
-            void InternalStartServer()
+            void InternalStartClient()
             {
-                Debug.Log($"Server is starting to listening at {ServerAddress}...\n");
-                try
-                {
-                    if (NetworkManager.Singleton.StartServer())
-                    {
-                        StartUp();
-                    }
-                    else
-                    {
-                        throw new CannotStartServer($"Could not start server {ServerAddress}");
-                    }
-                }
-                catch (Exception exception)
+                Debug.Log($"Client is trying to connect to server {ServerAddress}...\n");
+
+                NetworkManager.Singleton.NetworkConfig.ConnectionData = Encoding.ASCII.GetBytes(RoomPassword);
+                NetworkManager.Singleton.OnClientConnectedCallback += OnClientConnectedCallback;
+                NetworkManager.Singleton.OnClientDisconnectCallback += OnClientDisconnectCallback;
+
+                if (NetworkManager.Singleton.StartClient())
                 {
-                    callBack(false, exception.Message);
-                    throw;
+                    InitializeGame();
                 }
-                callBack(true, $"Server is listening at {ServerAddress}.");
             }
         }
 
+        /// <summary>
+        /// The maximal waiting time in seconds a client is willing to wait until a connection
+        /// can be established.
+        /// </summary>
+        private const float maxWaitingTime = 5 * 60;
```

Comment:
Missing units. (If the unit is seconds, this also seems like a very long time to wait.)